uniffi_bindgen/src/bindings/gecko_js/gen_gecko_js.rs

```diff
@@ -0,0 +1,463 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use anyhow::Result;
+use askama::Template;
+use heck::{CamelCase, MixedCase};
+use serde::{Deserialize, Serialize};
+
+use crate::interface::*;
+use crate::MergeWith;
+
+use super::namespace_to_file_name;
+use super::webidl::{
+    BindingArgument, BindingFunction, ReturnBy, ReturningBindingFunction, ThrowBy,
+};
+
+/// Config options for the generated Firefox front-end bindings. Note that this
+/// can only be used to control details *that do not affect the underlying
+/// component*, since the details of the underlying component are entirely
+/// determined by the `ComponentInterface`.
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct Config {
+    // ...
```

Comment:
Hmm, good question! There's limited customization that we can do, just because `Codegen.py` expects files and C++ namespaces to be laid out a certain way. I could imagine letting consumers specify a prefix, to make the WebIDL interface, namespace, dictionary, and type names globally unique, though—can you think of others we might want now?